web/react/components/create_comment.jsx

```diff
@@ -205,6 +206,11 @@ module.exports = React.createClass({
             );
         }
 
+        var extraInfo = <MsgTyping channelId={this.props.channelId} parentId={this.props.rootId}  />;
+        if (this.state.messageText.split(' ').length > 1 && allowTextFormatting) {
```

Comment:
I'm pretty sure the spec was written that way as to keep the text box area 'clean' when nothing is going on (e.g. someone else typing or you typing).  Currently the issue of having it up all the time is that then the 'Someone is typing...' message would conflict/never appear.  However if we wanted to we could switch the priorities (have the help text up all the time and replace with 'someone is typing' when that occurs) or just move the help text to the right side of the text box (slack does this, but it also doesn't show the help text until a certain amount has been typed as well).